v2.0/performance-tuning.md

```diff
@@ -0,0 +1,1275 @@
+
+<img src="{{ 'images/v2.0/perf_tuning_concepts4.png' | relative_url }}" alt="Perf tuning concepts" style="max-width:100%" />
+
+## Single-region deployment
+
+<!-- roachprod instructions for single-region deployment
+1. Reserve 4 VMs: roachprod create jesse-crdb -n 4 --local-ssd
+2. Optimize the local SSDs for write performance: roachprod run jesse-crdb:1-3 -- 'sudo umount /mnt/data1; sudo mount -o discard,defaults,nobarrier /dev/disk/by-id/google-local-ssd-0 /mnt/data1/; mount | grep /mnt/data1'
+3. Put cockroach` on all VM: `roachprod run jesse-crdb "curl https://binaries.cockroachdb.com/cockroach-v2.0.3.linux-amd64.tgz | tar -xvz; mv cockroach-v2.0.3.linux-amd64/cockroach cockroach"
+4. Start the cluster: roachprod start jesse-crdb:1-3
+5. List the IP address of all instances: roachprod list -d jesse-crdb
+6. SSH onto instance 4: roachprod run jesse-crdb:4
+7. Start sql shell: ./cockroach sql --insecure --host=<IP address of any node>
+8. Run the SQL commands in Step 4 below.
+9. Still on instance 4, install the Python client. See step 5.
+-->
+
+### Step 1. Configure your network
+
+CockroachDB requires TCP communication on two ports:
+
+- **26257** (`tcp:26257`) for inter-node communication (i.e., working as a cluster)
+- **8080** (`tcp:8080`) for accessing the Admin UI
+
+Since GCE instances communicate on their internal IP addresses by default, you don't need to take any action to enable inter-node communication. However, if you want to access the Admin UI from your local network, you must [create a firewall rule for your project](https://cloud.google.com/vpc/docs/using-firewalls):
+
+Field | Recommended Value
+------|------------------
+Name | **cockroachadmin**
+Source filter | IP ranges
+Source IP ranges | Your local network's IP ranges
+Allowed protocols | **tcp:8080**
+Target tags | `cockroachdb`
+
+{{site.data.alerts.callout_info}}
+The **tag** feature will let you easily apply the rule to your instances.
+{{site.data.alerts.end}}
+
+### Step 2. Create instances
+
+1. [Create 3 instances](https://cloud.google.com/compute/docs/instances/create-start-instance) for your CockroachDB nodes. While creating each instance:  
+    - Select the **us-east1-b** [zone](https://cloud.google.com/compute/docs/regions-zones/).
+    - Use the `n1-standard-4` machine type (4 vCPUs, 15 GB memory).
+    - [Create and mount a local SSD](https://cloud.google.com/compute/docs/disks/local-ssd#create_local_ssd).
+    - To apply the Admin UI firewall rule you created earlier, click **Management, disk, networking, SSH keys**, select the **Networking** tab, and then enter `cockroachdb` in the **Network tags** field.
+
+2. Note the internal IP address of each `n1-standard-4` instance. You'll need these addresses when starting the CockroachDB nodes.
+
+3. SSH to each instance and [optimize the local SSD for write performance](https://cloud.google.com/compute/docs/disks/performance#optimize_local_ssd) (see the **Disable write cache flushing** section).
```

Comment:
Hmm, I assumed this was safe since it's a technique that we're featuring in our tpc-c benchmarking. @nvanbenschoten, @bdarnell, can either of you confirm that this is in fact safe and, if so, help me come up with language that clarifies that fact?

Reply:
Also, just to be clear, @knz, the reason I'm featuring this is, based on my testing, it significantly improves performance. @robert-s-lee is also impressed with the speedup it gives you.

Reply:
cc. @petermattis and @arjunravinarayan. I think they'll be able to give the most accurate answers to the following questions: "is it safe?" and "should we recommend it?".

Reply:
I believe it is not safe to do on cloud instances. It is what one would do to replicate the performance characteristics of an on-prem capacitor-backed SSDs would give you, and this is why we have these instructions in our performance whitepaper. But I do not believe it is suitable guidance for a performance tuning guide in general.

Reply:
OK, thanks, @arjunravinarayan. If @petermattis agrees that this is not safe to do on cloud instances, I'll remove this and rerun all the numbers.